runtime/gc_glue_java/EnvironmentDelegate.hpp

```diff
@@ -216,15 +216,21 @@ class MM_EnvironmentDelegate
 		}
 
 		if (_extensions->objectModel.isIndexable(destinationObjectPtr)) {
-			/* Updates internal field of indexable objects. Every indexable object have an extra field
-			 * that can be used to store any extra information about the indexable object. One use case is
-			 * OpenJ9 where we use this field to point to array data. In this case it will always point to
-			 * the address right after the header, in case of contiguous data it will point to the data
-			 * itself, and in case of discontiguous arraylet it will point to the first arrayiod. How to
-			 * updated dataAddr is up to the target language that must override fixupDataAddr */
-			_extensions->indexableObjectModel.fixupDataAddr(destinationObjectPtr);
-
-			if (_extensions->isVLHGC()) {
+#if defined(J9VM_ENV_DATA64)
+			if (_vmThread->isIndexableDataAddrPresent) {
+				/**
+				 * Updates internal field of indexable objects. Every indexable object have an extra field
+				 * that can be used to store any extra information about the indexable object. One use case is
+				 * OpenJ9 where we use this field to point to array data. In this case it will always point to
+				 * the address right after the header, in case of contiguous data it will point to the data
+				 * itself, and in case of discontiguous arraylet it will point to the first arrayiod. How to
+				 * updated dataAddr is up to the target language that must override fixupDataAddr
+				 */
+				_extensions->indexableObjectModel.fixupDataAddr(destinationObjectPtr);
+			}
+#endif /* J9VM_ENV_DATA64 */
+
+			if (UDATA_MAX != _extensions->getOmrVM()->_arrayletLeafSize) {
```

Comment:
Just a note for future changes....
Once we introduced off-heap, we should probably add another check here: 'if off heap disabled', since off-heap will also have leaf size set, but will not have leaf pointers in the header (if I remember correctly).

On the other side, I believe none of the 2 checks (if leaf size set and if off-heap disabled) are really necessary, since the call should attempt to do something only if the shape of object is discontinuous and it will be possible to encounter only if leaf size is set and off heap disabled. But this code path is not so perf critical, so for clarity it's better to have both checks explicit.